waspc/src/Wasp/Project/ExternalConfig/TsConfig.hs

```diff
@@ -0,0 +1,42 @@
+module Wasp.Project.ExternalConfig.TsConfig
+  ( analyzeTsConfigFile,
+  )
+where
+
+import Control.Monad.Except (ExceptT (ExceptT), runExceptT)
+import qualified Data.ByteString.Lazy.UTF8 as BS
+import Data.Either.Extra (maybeToEither)
+import StrongPath (Abs, Dir, File, Path', toFilePath)
+import qualified Wasp.ExternalConfig.TsConfig as T
+import Wasp.Generator.ExternalConfig.TsConfig (validateTsConfig)
+import Wasp.Project.Common
+  ( TsConfigFile,
+    WaspProjectDir,
+    findFileInWaspProjectDir,
+    tsConfigInWaspProjectDir,
+  )
+import Wasp.Util (validateToEither)
+import qualified Wasp.Util.IO as IOUtil
+import Wasp.Util.Json (parseJsonWithComments)
+
+analyzeTsConfigFile :: Path' Abs (Dir WaspProjectDir) -> IO (Either [String] T.TsConfig)
+analyzeTsConfigFile waspDir = runExceptT $ do
+  tsConfigFile <- ExceptT findTsConfigOrError
+  tsConfig <- ExceptT $ readTsConfigFile tsConfigFile
+  ExceptT $ validateToEither validateTsConfig tsConfig
+  where
+    findTsConfigOrError = maybeToEither [fileNotFoundMessage] <$> findTsConfigFile waspDir
+    fileNotFoundMessage = "Couldn't find the tsconfig.json file in the " ++ toFilePath waspDir ++ " directory"
+
+findTsConfigFile :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs (File TsConfigFile)))
+findTsConfigFile waspProjectDir = findFileInWaspProjectDir waspProjectDir tsConfigInWaspProjectDir
+
+readTsConfigFile :: Path' Abs (File TsConfigFile) -> IO (Either [String] T.TsConfig)
+readTsConfigFile tsConfigFile = do
+  tsConfigContent <- IOUtil.readFileBytes tsConfigFile
+
+  parseResult <- parseJsonWithComments . BS.toString $ tsConfigContent
+
+  case parseResult of
+    Right tsConfig -> return $ Right tsConfig
+    Left err -> return $ Left ["Failed to parse tsconfig.json file: " ++ err]
```

Comment:
You could combine these two into just one function, to make it a bit simpler, there doesn't seem to be any need to have `findTsConfigFile` as a separate step.

You could probably even completely ditch "finding" of the file and just go for the reading of it, and if there is no file, reading will fail anyway. But maybe this way we get nicer error message? Yeah, that might be so, not sure if worth it though.

I am just looking how to simplify this file, it seems like it has a lot of code but doesn't do much. But maybe that is ok.

Reply:
I feel like it's simple enough - the finding of the file produces a `Maybe` and this way I can handle it at the top level vs. handling it inside of `readTsConfigFile` 🤷

Reply:
Ok, let's leave it then!